src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -457,51 +456,42 @@ private static void ValidateAccessor(Expression instance, MethodInfo method, Par
 
         private static void ValidateAccessorArgumentTypes(MethodInfo method, ParameterInfo[] indexes, ref ReadOnlyCollection<Expression> arguments)
         {
-            if (indexes.Length > 0)
+            if (indexes.Length != arguments.Count)
```

Comment:
You're probably right; I've traced down where this method is being called from (self-note: `Expression.Property` with something returning not-null for `GetGetMethod`). Will add the corresponding tests tomorrow.